### PR TITLE
Docs dateDiff

### PR DESCRIPTION
**Pull request by godfreyd**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Category (leave one):
- Documentation

### CODE REVIEW

File: docs/en/query_language/functions/date_time_functions.md [hunk 1]
```diff
@@ -311,22 +311,58 @@ SELECT
...
+dateDiff('unit', startdate, enddate, [timezone])
+```
+
+**Parameters**
+
+- `unit` — Time unit, in which the returned value is expressed. [String](../syntax.md#syntax-string-literal). 
+
+    Supported values:
+
+    | unit   |
```
Comment (blinkov) [inline docs/en/query_language/functions/date_time_functions.md]: why not just plain list?

File: docs/en/query_language/functions/date_time_functions.md [hunk 2]
```diff
@@ -311,22 +311,58 @@ SELECT
...
-|minute  |
-|hour    |
-|day     |
-|week    |
-|month   |
-|quarter |
-|year    |
+## dateDiff {#datediff}
+
+Returns the difference between two time values.
```
Comment (blinkov) [inline docs/en/query_language/functions/date_time_functions.md]:
```suggestion
Returns the difference between two Date or DateTime values.
```

File: docs/en/query_language/functions/date_time_functions.md [hunk 3]
```diff
@@ -311,22 +311,58 @@ SELECT
...
+    |quarter |
+    |year    |
+
+- `startdate` — The first time value to compare. [Date](../../data_types/date.md) or [DateTime](../../data_types/datetime.md).
+- `enddate` — The second time value to compare. [Date](../../data_types/date.md) or [DateTime](../../data_types/datetime.md).
+- `timezone` — Optional parameter. If specified, it is applied to both `startdate` and `enddate`. If not specified, timezones of `startdate` and `enddate` are used. If they are not the same, the result is unspecified. [Time Zone](../../data_types/datetime.md#time-zones).
+
+**Returned value**
+
+Difference between `startdate` and `enddate` expressed in `unit`.
```
Comment (blinkov) [inline docs/en/query_language/functions/date_time_functions.md]: Needs clarifications:
* What is the data time of the result?
* How fractional results are handled? Rounding?
* What happens if startdate is later than enddate. Negative result? Absolute difference? Error?

File: docs/en/query_language/functions/date_time_functions.md [hunk 4]
```diff
@@ -311,22 +311,58 @@ SELECT
...
+    |hour    |
+    |day     |
+    |week    |
+    |month   |
+    |quarter |
+    |year    |
+
+- `startdate` — The first time value to compare. [Date](../../data_types/date.md) or [DateTime](../../data_types/datetime.md).
+- `enddate` — The second time value to compare. [Date](../../data_types/date.md) or [DateTime](../../data_types/datetime.md).
+- `timezone` — Optional parameter. If specified, it is applied to both `startdate` and `enddate`. If not specified, timezones of `startdate` and `enddate` are used. If they are not the same, the result is unspecified. [Time Zone](../../data_types/datetime.md#time-zones).
```
Comment (blinkov) [inline docs/en/query_language/functions/date_time_functions.md]: needs separate example, if timezones are forced to be the same, why do they affect the result?

File: docs/en/query_language/functions/date_time_functions.md [hunk 5]
```diff
@@ -311,22 +311,58 @@ SELECT
...
+    Supported values:
+
+    | unit   |
+    | ------ |
+    |second  |
+    |minute  |
+    |hour    |
+    |day     |
+    |week    |
+    |month   |
```
Comment (blinkov) [inline docs/en/query_language/functions/date_time_functions.md]: Needs detailed explanation on how exactly does the months and beyond work.
For example, difference between December 31 and January 1 is one month or zero?

**Review by blinkov — CHANGES_REQUESTED** [review]

**Comment by alexey-milovidov** [comment]
No chance this PR will be finished.

**Pull request merged**